Services/Common/OJS.Services.Common.Data/Implementations/SubmissionsForProcessingCommonDataService.cs

```diff
@@ -57,18 +58,15 @@ public async Task Add(int submissionId)
         };
 
         await this.Add(submissionForProcessing);
+
+        return submissionForProcessing;
     }
 
-    public async Task<SubmissionForProcessing> AddOrUpdate(int submissionId)
+    public async Task<SubmissionForProcessing> CreateIfNotExists(int submissionId)
```

Comment:
Why did you change the name of the method? Doesn't it do exactly Adding or Updating, and not just Creating if does not exist?

Reply:
This logic was modified with pr:  https://github.com/SoftUni-Internal/OpenJudgeSystem/pull/819. I am reverting the previous name.